packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -221,4 +223,28 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
         }
     }
 
+    /**
+     * Get the formatting options to be used when calling `jsoncparser`.
+     * The formatting options are based on the corresponding preference values.
+     * @param uri the preference settings URI.
+     *
+     * @returns a tuple representing the tab indentation size, and if it is spaces.
+     */
+    protected getFormattingOptions(uri?: string): [number, boolean] {
+        // Get the global `tabSize` preference value. Defaults to 2.
+        const globalTabSize = this.preferenceService.get('editor.tabSize', 2, uri);
```

Comment:
it should not be necessary, if there are not `[json].editor.tabSize` then the preference service will try `editor.tabSize` as well. Clients should not care about reimplementing overriding logic, It's a responsibility of `PreferenceService`.

Reply:
It did not seem to work well initially. I noticed that if the global `editor.insertSpaces` was updated to false and no `[json].insertSpaces` was user defined, the value remained true (which caused incorrect indentation). In this way we explicitly fallback to the values we care about.

Reply:
@akosyakov during testing I noticed an odd behavior that made me conclude it might be necessary or better to fallback to the values at a higher scope. For example, when updating the preference `editor.insertSpaces` to false and no user preference `[json].editor.insertSpaces` was defined, the global preference was never read which lead to an odd behavior. Instead, the fallback used to be always `true`.

Reply:
It sounds like a bug, I would try to debug later.